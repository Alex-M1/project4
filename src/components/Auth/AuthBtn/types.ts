export interface IAuthBtn {
    isRedirect: boolean;
    setIsRedirect: (isRedirect: boolean) => void;
    signInRequest: () => void;
}
