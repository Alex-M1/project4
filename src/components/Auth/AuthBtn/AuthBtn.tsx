import React from 'react';

import Button from '../../_common_/Button';
import { IAuthBtn } from './types';
import { StInputDiv } from '../AuthIpt/styled';

const AuthBtn = ({ signInRequest }: IAuthBtn) => {
    const onButtonClick = () => signInRequest();
    return (
        <StInputDiv>
            <Button
                text="signInBtn"
                onClick={onButtonClick}
            />
        </StInputDiv>
    );
};

export default AuthBtn;
