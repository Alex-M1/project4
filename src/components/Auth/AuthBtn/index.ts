import { connect } from 'react-redux';
import { getIsRedirect } from 'store/user/selectors';
import { signInRequest, setIsRedirect } from 'store/user/action';
import { AppStateType } from 'store/rootReducer';
import AuthBtn from './AuthBtn';

const mapStateToProps = (state: AppStateType) => ({
    isRedirect: getIsRedirect(state),
});
const mapDispatchToProps = {
    signInRequest,
    setIsRedirect,
};

export default connect(mapStateToProps, mapDispatchToProps)(AuthBtn);
