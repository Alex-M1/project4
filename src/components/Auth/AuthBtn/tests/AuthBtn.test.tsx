import React from 'react';
import { mount, shallow } from 'enzyme';
import AuthBtn from '../AuthBtn';

jest.mock('react-i18next', () => ({
    useTranslation: jest.fn().mockImplementation(() => ({
      t: jest.fn(),
    })),
  }),
);

describe('AuthBtn', () => {
    let props;
    beforeEach(() => {
        props = {
            signInRequest: jest.fn(),
        };
    });
    it('Should match snapshot', () => {
        const component = shallow(<AuthBtn {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StInputDiv', () => {
        const component = shallow(<AuthBtn {...props}/>);
        expect(component.find('styled__StInputDiv')).toHaveLength(1);
    });
    it('should render Button', () => {
        const component = shallow(<AuthBtn {...props}/>);
        expect(component.find('Button')).toHaveLength(1);
    });
    it('should call onClick', () => {
        const component = mount(<AuthBtn {...props}/>);
        component.find('Button').simulate('click');
        expect(component.props().signInRequest).toHaveBeenCalled();
    });
});
