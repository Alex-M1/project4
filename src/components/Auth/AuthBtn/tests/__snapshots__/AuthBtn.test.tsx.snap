// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthBtn Should match snapshot 1`] = `"<div class=\\"styled__StInputDiv-iTvJvh gqdFEH\\"><button type=\\"button\\" class=\\"styled__StButton-iwFefC jJqhBl\\"></button></div>"`;

exports[`Button Should match snapshot 1`] = `"<button type=\\"button\\" class=\\"styled__StButton-iwFefC jJqhBl\\"></button>"`;
